rs/replicated_state/src/canister_state/queues/queue.rs

```diff
@@ -181,6 +181,15 @@ impl CanisterQueue {
         Ok(())
     }
 
+    /// Releases a reserved response slot.
+    ///
+    /// This is used when a request in the reverse queue is dropped before having
+    /// had a chance to be popped.
+    pub(super) fn release_reserved_response_slot(&mut self) {
+        debug_assert!(self.response_slots > 0);
+        self.response_slots -= 1;
```

Comment:
Is underflow better than crash in production?

Reply:
Good point. I guess this should be a `saturating_sub()` at most.

We seem to have strong reservations against `asserts` in production code, so it's less contentious to go with a `debug_assert` instead. But underflow would be quite bad in itself, so we should do something more than blindly subtracting here. Maybe forcing the caller to invoke `check_has_reserved_response_slot()` first and then having a hard `assert` here is also an option.